torchvision/transforms/v2/_utils.py

```diff
@@ -93,3 +96,62 @@ def _check_padding_arg(padding: Union[int, Sequence[int]]) -> None:
 def _check_padding_mode_arg(padding_mode: Literal["constant", "edge", "reflect", "symmetric"]) -> None:
     if padding_mode not in ["constant", "edge", "reflect", "symmetric"]:
         raise ValueError("Padding mode should be either constant, edge, reflect or symmetric")
+
+
+def _find_labels_default_heuristic(inputs: Any) -> torch.Tensor:
+    """
+    This heuristic covers three cases:
+
+    1. The input is two-tuple whose second item is a labels tensor. This happens for already batched classification
+       inputs for Mixup and Cutmix.
+    2. The input is a two-tuple whose second item is a dictionary that contains the labels tensor under a label-like
+       (see below) key. This happens for the inputs of detection models.
+    3. The input is a dictionary that is structured as the one from 2.
+
+    What is "label-like" key? We first search for an case-insensitive match of 'labels' inside the keys of the
+    dictionary. This is the name our detection models expect. If we can't find that, we look for a case-insensitive
+    match of the term 'label' anywhere inside the key, i.e. 'FooLaBeLBar'. If we can't find that either, the dictionary
+    contains no "label-like" key.
+
+    """
+    # TODO: Document list and why
+    if isinstance(inputs, (tuple, list)):
+        inputs = inputs[1]
+
+    # Mixup, Cutmix
+    if isinstance(inputs, torch.Tensor):
+        return inputs
+
+    if not isinstance(inputs, collections.abc.Mapping):
+        raise ValueError(
+            f"When using the default labels_getter, the input passed to forward must be a dictionary or a two-tuple "
+            f"whose second item is a dictionary or a tensor, but got {inputs} instead."
+        )
+
+    # Tries to find a "labels" key, otherwise tries for the first key that contains "label" - case insensitive
```

Comment:
Nit: can we remove this comment now that we have detailed description in the docstring?